openwisp_firmware_upgrader/tests/test_tasks.py

```diff
@@ -42,3 +42,27 @@ def test_batch_upgrade_timeout(self, *args):
         self.assertEqual(BatchUpgradeOperation.objects.count(), 1)
         batch = BatchUpgradeOperation.objects.first()
         self.assertEqual(batch.status, 'failed')
+
+    @mock.patch(_mock_upgrade, return_value=True)
+    @mock.patch(_mock_connect, return_value=True)
+    @mock.patch('logging.Logger.warning')
+    def test_upgrade_firmware_resilience(self, mocked_logger, *args):
+        device_fw = self._create_device_firmware(upgrade=True)
+        upgrade_op = device_fw.image.upgradeoperation_set.first()
+        upgrade_op.delete()
+        tasks.upgrade_firmware.run(upgrade_op.id)
+        mocked_logger.assert_called_with(
+            f'The UpgradeOperation object with id {upgrade_op.id} has been deleted'
+        )
+
+    @mock.patch(_mock_upgrade, return_value=True)
+    @mock.patch(_mock_connect, return_value=True)
+    @mock.patch('logging.Logger.warning')
+    def test_batch_upgrade_operation_resilience(self, mocked_logger, *args):
+        env = self._create_upgrade_env()
+        batch = BatchUpgradeOperation.objects.create(build=env['build2'])
+        batch.delete()
```

Comment:
there's no need to actually create and delete the object, you can simply instantiate it to get the uuid: `BatchUpgradeOperation().pk`, this makes the test faster to run, which is good because `TransactionTestCase` is slow.